codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpRpcProtocolGenerator.java

```diff
@@ -0,0 +1,329 @@
+    @Override
+    public void generateResponseDeserializers(GenerationContext context) {
+        TopDownIndex topDownIndex = context.getModel().getKnowledge(TopDownIndex.class);
+        Set<OperationShape> containedOperations = new TreeSet<>(
+                topDownIndex.getContainedOperations(context.getService()));
+        for (OperationShape operation : containedOperations) {
+            generateOperationDeserializer(context, operation);
+        }
+    }
+
+    private void generateOperationDeserializer(GenerationContext context, OperationShape operation) {
+        GoStackStepMiddlewareGenerator middleware = GoStackStepMiddlewareGenerator.createDeserializeStepMiddleware(
+                ProtocolGenerator.getDeserializeMiddlewareName(operation.getId(), getProtocolName()),
+                ProtocolUtils.OPERATION_DESERIALIZER_MIDDLEWARE_ID);
+
+        SymbolProvider symbolProvider = context.getSymbolProvider();
+        Model model = context.getModel();
+        GoWriter writer = context.getWriter();
+        StructureShape outputShape = ProtocolUtils.expectOutput(context.getModel(), operation);
+        Symbol outputSymbol = symbolProvider.toSymbol(outputShape);
+        ApplicationProtocol applicationProtocol = getApplicationProtocol();
+        Symbol responseType = applicationProtocol.getResponseType();
+        String errorFunctionName = ProtocolGenerator.getOperationErrorDeserFunctionName(
+                operation, context.getProtocolName());
+
+        middleware.writeMiddleware(writer, (generator, w) -> {
+            writer.addUseImports(SmithyGoDependency.FMT);
+            writer.addUseImports(SmithyGoDependency.SMITHY);
+
+            writer.write("out, metadata, err = next.$L(ctx, in)", generator.getHandleMethodName());
+            writer.write("if err != nil { return out, metadata, err }");
+            writer.write("");
+
+            writer.write("response, ok := out.RawResponse.($P)", responseType);
+            writer.openBlock("if !ok {", "}", () -> {
+                writer.write(String.format("return out, metadata, &smithy.DeserializationError{Err: %s}",
+                        "fmt.Errorf(\"unknown transport type %T\", out.RawResponse)"));
+            });
+            writer.write("");
+
+            writer.openBlock("if response.StatusCode < 200 || response.StatusCode >= 300 {", "}", () -> {
+                writer.write("return out, metadata, $L(response)", errorFunctionName);
+            });
+
+            writer.write("output := &$T{}", outputSymbol);
+            writer.write("out.Result = output");
+            writer.write("");
+
+            deserializeOutputDocument(context, operation);
+            deserializingDocumentShapes.add(ProtocolUtils.expectOutput(model, operation));
+            writer.write("");
+
+            writer.write("return out, metadata, err");
+        });
+        writer.write("");
+
+        Set<StructureShape> errorShapes = HttpProtocolGeneratorUtils.generateErrorDispatcher(
```

Comment:
It doesn't look like the error dispatcher is handling the private `isErrorCodeInBody` property that's used for some implementing protocols. That property just looks set on the constructor and not utilized.

Reply:
I'm not sure if it matters here. Unlike TS, the go dispatcher is never attempting to parse the error body, only copying it into a seekable type. If it happens that it's empty, nothing changes. Perhaps we should just get rid of the property

Reply:
Minor context: Go v1 apparently would preparse (json) into an empty interface map, but that was fairly memory intensive.